### PR TITLE
WIP: Manual Github Job to communicate with Library owners

### PR DESCRIPTION
**Pull request by fbm3307**
This PR is to automate the communication with the imagestreams/templates owners (list in library/official.yaml) by creating github issues on those repos whenever the github workflow/job is triggered

Signed-off-by: Feny Mehta <fbm3307@gmail.com>

### CODE REVIEW

**Comment by fbm3307** [comment]
/do-not-merge/work-in-progress

**Review by mmulholla — COMMENTED** [review]
Added some comments

File: scripts/create_issue.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+def _make_gihub_request(method="post", uri="issues", body=None, params={}, headers={}, verbose=False, repo=""):
+    GITHUB_BASE_URL = "https://api.github.com"
+    headers.update({"Authorization": f'Bearer {endpoint_data["access_token"]}',
+                    "Accept": "application/vnd.github.v3+json"})    
+    print(headers)
+    url = f'{GITHUB_BASE_URL}/repos/{repo}/{uri}'
+
+    print(f"API url: {url}")
+    method_map = {"get": requests.get,
```
[the remark this answers is not in the document]
Comment (fbm3307) [reply scripts/create_issue.py:28]: done

File: .github/workflows/autocommunicate.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+          python3 -m venv ve1
+          cd scripts && ../ve1/bin/pip3 install -r requirements.txt && cd ..
+          cd scripts && ../ve1/bin/python3 setup.py install && cd ..
+      
+      - name: Auto Communicate
+        id: auto_communicate
+        env:
+          github_token: ${{secrets.GITHUB_TOKEN}}
+        run: |
+          if [[ "${{github.event_name}}" == "workflow_dispatch" ]]; then
```
[the remark this answers is not in the document]
Comment (fbm3307) [reply .github/workflows/autocommunicate.yml:36]: okay

File: scripts/library.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    for reponame in githubcontent:
+        imagestreamLocationSet = set() #Initialize the locationSet
+        if("imagestreams" in githubcontent[reponame]):
+            #code for imagestream
+            for ele in githubcontent[reponame]["imagestreams"]:
+                location = ele["location"]
+                temp = (str(location).split("//")[1]).split("/")
+                repo1,repo2 = temp[1], temp[2]
+                #finalUrl = "https://github.com/" + str(repo1) +"/" + str(repo2)
+                finalUrl = str(repo1) + "/" + str(repo2)
```
[the remark this answers is not in the document]
Comment (fbm3307) [reply scripts/library.py:46]: okay

File: scripts/library.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        targetDict = testDict
+    else:
+        print("Invalid input")
+        exit()
+    for repoName in targetDict.keys():
+        repoList = targetDict[repoName]
+        for repo in repoList:
+            create_an_issue(title="sample issue",description="sample description", repo=str(repo))
+            input("Check for the issue")
+else:
```
[the remark this answers is not in the document]
Comment (fbm3307) [reply scripts/library.py:82]: got it

File: scripts/library.py [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+except ImportError:
+    from yaml import Loader, Dumper
+
+
+imageStreamDict = {}
+templateDict = {}
+combinedDict = {}
+testDict = {"test":["fbm3307/test-learn"]}
+
+def load_yaml():
```
[the remark this answers is not in the document]
Comment (fbm3307) [reply scripts/library.py:29]: okay

**Comment by fbm3307** [comment]
This PR was raised in another repo, hence not required here . Closing it

**Pull request closed without merging**